flax/core/meta.py

```diff
@@ -304,7 +308,17 @@ def get_partition_spec(tree: Any) -> Any:
   def f(x):
     if isinstance(x, Partitioned):
       return x.get_partition_spec()
+    # Unboxed arrays, which should be replicated across all devices
+    elif hasattr(x, 'shape'):
+      return jax.sharding.PartitionSpec()
     else:
       return None
   return jax.tree_map(f, tree,
                       is_leaf=lambda x: isinstance(x, Partitioned))
+
+
+def get_sharding(tree: Any, mesh: jax.sharding.Mesh) -> Any:
+  """Extracts a jax.sharding tree from a PyTree containing ``Partitioned`` values and a mesh."""
+  pspec_tree = get_partition_spec(tree)
+  return jax.tree_map(lambda x: jax.sharding.NamedSharding(mesh, x), pspec_tree,
+                      is_leaf=lambda x: isinstance(x, Partitioned))
```

Comment:
`is_leaf` not needed here since `get_partition_spec` maps all `Partitioned` instances to `PartitionSpec`.
```suggestion
  pspec_tree = get_partition_spec(tree)
  return jax.tree_map(lambda x: jax.sharding.NamedSharding(mesh, x), pspec_tree)
```

Reply:
Done.